File: leetcode/src/PlusOne.java
import java.util.Arrays;

public class PlusOne {
    public static void main(String[] args) {
        PlusOne instance = new PlusOne();
        int[] result = instance.plusOne(new int[]{1,2,3});
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + ",");
        }
        System.out.println();
        result = instance.plusOne(new int[]{4,3,2,1});
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + ",");
        }
        System.out.println();
        result = instance.plusOne(new int[]{9});
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + ",");
        }
        System.out.println();
    }
    public int[] plusOne(int[] digits) {
        int[] result = digits;
        int plus = 1;
        for (int i = digits.length - 1; i >= 0; i--) {
            int num = digits[i] + plus;
            if (num >= 10) {
                plus = num / 10;
                digits[i] = num % 10;
            } else {
                plus = 0;
                digits[i] = num;
            }
        }
        if (plus > 0) {
            result = new int[digits.length + 1];
            result[0] = 1;
            for (int i = 0; i < digits.length; i++) {
                result[i + 1] = digits[i];
            }
        }
        return result;
    }
}

File: leetcode/src/python/plus_one.py
from typing import List

class PlusOne:
    def plusOne(self, digits: List[int]) -> List[int]:
        left = 1
        for i in range(len(digits) - 1, -1, -1):
            if left == 0:
                break
            else:
                sum = digits[i] + left
                if sum < 10:
                    digits[i] = sum
                    left = 0
                else:
                    digits[i] = 0
                    left = 1
        if left == 1:
            digits.insert(0, 1)
        return digits

if __name__ == '__main__':
    instance = PlusOne()
    digits = [9]
    print(instance.plusOne(digits))
